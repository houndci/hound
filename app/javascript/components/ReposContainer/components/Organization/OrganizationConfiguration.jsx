import React from "react";
import classNames from "classnames";
import $ from "jquery";

import "object-assign-shim";

import * as Ajax from "../../../../modules/Ajax";

export default class OrganizationConfiguration extends React.Component {
  constructor(props) {
    super(props);

    let repo = "";

    if (props.repo) {
      repo = props.repo;
    }

    this.state = {
      configUpdated: false,
      enabled: !!props.enabled,
      repo
    };
  }

  setEnabled(enabled) {
    this.setState({ enabled });
    this.updateOwner({ config_enabled: enabled });
  }

  setRepo(repo) {
    this.setState({ repo });
    this.updateOwner({ config_repo: repo });
  }

  handleChange(event) {
    this.setRepo(event.target.value);
  }

  toggle() {
    const enabled = !this.state.enabled;
    this.setEnabled(enabled);
  }

  updateOwner(options) {
    const currentState = {
      config_enabled: this.state.enabled,
      config_repo: this.state.repo
    };

    this.setState({ configUpdated: false });
    Ajax.updateOwner(this.props.id, { ...currentState, ...options }).then(() =>
      this.setState({ configUpdated: true })
    );
  }

  renderRepoSelect() {
    if (this.state.enabled) {
      return (
        <div className="organization-header-source">
          <span className="organization-header-label">Use .hound.yml from</span>
          <select
            className="organization-header-select"
            onChange={event => this.handleChange(event)}
            value={this.state.repo}
          >
            {this.renderRepoOptions()}
          </select>
          <div className="inline-flash--success config-enabled">
            {this.state.configUpdated &&
              <span data-role="config-saved">&#10004;</span>}
          </div>
        </div>
      );
    } else {
      return null;
    }
  }

  renderRepoOptions() {
    return this.props.repos.map(repo =>
      <option key={repo.id} value={repo.name}>
        {repo.name}
      </option>
    );
  }

  render() {
    const switchId = `toggle-${this.props.id}`;
    return (
      <div className="organization-header-config">
        <div className="organization-header-toggle">
          <span className="organization-header-label">
            Use organization-wide config
          </span>

          <input
            id={switchId}
            checked={this.state.enabled}
            className="organization-header-toggle-input"
            onChange={() => this.toggle()}
            type="checkbox"
            name="toggle"
          />

          <label className="toggle-switch" htmlFor={switchId} />
        </div>

        {this.renderRepoSelect()}
      </div>
    );
  }
}
