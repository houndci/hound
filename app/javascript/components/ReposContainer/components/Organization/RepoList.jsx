import React from 'react';

import EmptyRepoList from './RepoList/EmptyRepoList';
import PopulatedRepoList from './RepoList/PopulatedRepoList';

export default class RepoList extends React.Component {
  render() {
    const {
      repos,
      onRepoClicked,
      isProcessingId,
      filterTerm,
    } = this.props;

    if (repos.length > 0) {
      return (
        <PopulatedRepoList
          repos={repos}
          onRepoClicked={onRepoClicked}
          isProcessingId={isProcessingId}
          filterTerm={filterTerm}
        />
      );
    } else {
      return (
        <EmptyRepoList />
      );
    }
  }
}
