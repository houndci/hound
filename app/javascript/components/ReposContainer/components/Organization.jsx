import React from 'react';

import OrganizationConfiguration from "./Organization/OrganizationConfiguration";
import RepoList from './Organization/RepoList';

export default class Organization extends React.Component {
  render() {
    const {
      name,
      onRepoClicked,
      isProcessingId,
      repos,
      configEnabled,
      configRepo,
      filterTerm,
      ownerId,
    } = this.props;

    return (
      <div className="organization" data-org-name={name}>
        <header className="organization-header">
          <h2 className="organization-header-title">{name}</h2>

          <OrganizationConfiguration
            enabled={configEnabled}
            id={ownerId}
            repo={configRepo}
            repos={repos}
          />
        </header>
        <section className="repo_listing">
          <RepoList
            repos={repos}
            onRepoClicked={onRepoClicked}
            isProcessingId={isProcessingId}
            filterTerm={filterTerm}
          />
        </section>
      </div>
    );
  }
}
