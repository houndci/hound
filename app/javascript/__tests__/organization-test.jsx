import Organization from "../components/ReposContainer/components/Organization";

it("renders an organization with ID appropriately (no repo processing)", () => {
  const org = {
    id: 1,
    name: "Test org"
  };
  const repos = [
    { id: 1, name: "Test repo" }
  ];

  const wrapper = shallow(
    <Organization
      name={org.name}
      key={org.id}
      repos={repos}
      onRepoClicked={jest.fn()}
      filterTerm={""}
      isProcessingId={null}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
