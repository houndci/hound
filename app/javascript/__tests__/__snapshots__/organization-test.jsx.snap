// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an organization with ID appropriately (no repo processing) 1`] = `
<div
  className="organization"
  data-org-name="Test org"
>
  <header
    className="organization-header"
  >
    <h2
      className="organization-header-title"
    >
      Test org
    </h2>
    <OrganizationConfiguration
      repos={
        Array [
          Object {
            "id": 1,
            "name": "Test repo",
          },
        ]
      }
    />
  </header>
  <section
    className="repo_listing"
  >
    <RepoList
      filterTerm=""
      isProcessingId={null}
      onRepoClicked={[MockFunction]}
      repos={
        Array [
          Object {
            "id": 1,
            "name": "Test repo",
          },
        ]
      }
    />
  </section>
</div>
`;
